crypto/evp/p5_crpt2.c

```diff
@@ -47,9 +48,10 @@ int PKCS5_PBKDF2_HMAC(const char *pass, int passlen,
         return 0;
     p = out;
     tkeylen = keylen;
-    if (!pass)
+    if (!pass) {
+        pass = empty;
         passlen = 0;
-    else if (passlen == -1)
+    } else if (passlen == -1)
```

Comment:
There might be some style tweaks needed, here.
OpenSSL style is usually to compare pointers against NULL, not to treat them as booleans.
I forget whether the rule about braces involves using braces for all branches of a conditional or none of them (as opposed to mixed, which is what this commit is doing).